temporalio/contrib/openai_agents/_mcp.py

```diff
@@ -0,0 +1,512 @@
+import asyncio
+import logging
+import uuid
+from datetime import timedelta
+from typing import Any, Callable, Optional, Sequence, Union
+
+from agents import AgentBase, RunContextWrapper
+from agents.mcp import MCPServer
+from mcp import GetPromptResult, ListPromptsResult  # type:ignore
+from mcp import Tool as MCPTool  # type:ignore
+from mcp.types import CallToolResult  # type:ignore
+
+from temporalio import activity, workflow
+from temporalio.api.enums.v1.workflow_pb2 import (
+    TIMEOUT_TYPE_HEARTBEAT,
+    TIMEOUT_TYPE_SCHEDULE_TO_START,
+)
+from temporalio.exceptions import ActivityError, ApplicationError
+from temporalio.worker import PollerBehaviorSimpleMaximum, Worker
+from temporalio.workflow import ActivityConfig, ActivityHandle
+
+logger = logging.getLogger(__name__)
+
+
+class StatelessTemporalMCPServer(MCPServer):
```

Comment:
So I think there is some confusion on the plugin/activity-provided instance of this and the workflow-author-instantiated instance of this. In theory the workflow author doesn't care about stateless/stateful or really anything but the name (and the task queue may end up being something we can implicitly obtain), it's more about what's registered at the plugin level. Something like `workflow.get_mcp_server_reference("foo")` may be cleaner. Can discuss off-PR next week.